xbmc/listproviders/DirectoryProvider.cpp

```diff
@@ -495,7 +496,17 @@ class CVideoSelectActionProcessor : public VIDEO::GUILIB::CVideoSelectActionProc
 
   bool OnPlaySelected() override
   {
-    ExecuteAction({"PlayMedia", m_item, "noresume"});
+    if (m_item.GetVideoInfoTag()->m_type == MediaTypeMovie)
+    {
+      auto videoItem = std::make_shared<CFileItem>(m_item);
+      CGUIDialogVideoVersion::PlayVideoVersion(videoItem, [](std::shared_ptr<CFileItem>& item)
+      {
+        ExecuteAction({"PlayMedia", *item.get(), "noresume"});
+      });
```

Comment:
This sort of logic and change sneaked in at several places. I think we should move all this to 

https://github.com/xbmc/xbmc/blob/master/xbmc/video/guilib/VideoPlayActionProcessor.h#L19 and/or
https://github.com/xbmc/xbmc/blob/master/xbmc/video/guilib/VideoSelectActionProcessor.h#L19

There is for example the "play from beginning"/"resume" user selection code centralized.

Reply:
As well as that, if you're navigating in the video node itself, rather than the library (so videos->movies->video versions->some version)  I'd much prefer to not open the dialog on a movie there as I've already navigated to 4K for example and picked the movie I want to watch.  Currently I have to choose the version again from the dialog.